department-of-veterans-affairs/caseflow
# Details: Correspondence and Appeals Tasks: Add Task Action: Other motion

Resolves [Details: Correspondence and Appeals Tasks: Add Task Action: Other motion](https://jira.devops.va.gov/browse/APPEALS-54933)

# Description
Added the following:
- Return to Inbound Ops route in the queue app
- Return to Inbound Ops Controller 
- Return to Inbound Ops option in the mail task drop down
- Return to Inbound Ops Modal 
- Three "reason for return" radio button options in the return to inbound ops modal
- A text field that appears when the "other" radio button is selected
- Method to create the new task and save reason for return
- Controller Spec test to verify the controller is functioning
- Jest tests to verify the front-end functionality
- Specs to verify back-end functionality

## Acceptance Criteria

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
You may need to set up some data first for this! To start:

1. Log in as an inbound ops team admin user
2. Navigate to /queue/correspondence/team and click on a correspondence
3. Under General Information, select "other motions" correspondence type and save changes.
4. Scroll down to the bottom and select "Create record"
5. Continue through step 1, and scroll to "tasks not related to appeal" on step 2.
6. Click "+ Add tasks", select an other motion task in the drop down, and add text in the textbox.
7. Click "Continue" from the second step. Before continuing, copy the link from the address bar to use later!
8.  On step 3, click the "submit" button. Then Confirm the submission. 

Now that the data is set up, you can continue to test the story. 
1. Log in as a Litigation Support user
2. Paste the address saved from the previous step, make sure to remove the last part of /review_package or /intake
3. This will take you to the Case details page for that appeal as the Lit Support user.
4. Click on the + button next to the "Tasks not related to appeal" section
5. In the Actions drop down next to the other motion task, verify the "Return to Inbound Ops Team" task is present
6. Click on the return to inbound ops, and the modal will appear.
7. The return button should initially not be clickable until a radio button is selected. 
8. Once a radio button is selected, the return button will be clickable
9. If "Other" is selected, the return button will remain locked until you type a reason for return.
10. Click on the "Return" button. Now, verify in the "Tasks not related" section that the success banner appears.
